src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,34 +1,84 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface NewMovieProps {
+  onAdd: (newMovie: Movie) => void;
+}
+
+export const NewMovie = ({ onAdd }: NewMovieProps) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isButtonDisabled =
+    !title.trim() || !imgUrl.trim() || !imdbUrl.trim() || !imdbId.trim();
+
+  const handleSubmit = function (event: React.FormEvent) {
+    event.preventDefault();
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
+    setCount(count + 1);
```

Comment:
```suggestion
    setCount(count => count + 1);
```
Always use callback to update state based on previous value

Reply:
sorry, can you explain why I need to do this? What is the difference?

Reply:
https://dev.to/csituma/when-do-you-use-setstate-with-a-callback-1f3g